tests/socketcall_32bit/test

```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/perl
+
+use strict;
+
+use Test;
+BEGIN { plan tests => 3 }
+
+use File::Temp qw/ tempfile /;
+
+my $basedir = $0;
+$basedir =~ s|(.*)/[^/]*|$1|;
+
+###
+# functions
+
+sub key_gen {
+	my @chars = ("A".."Z", "a".."z");
+	my $key = "testsuite-" . time . "-";
+	$key .= $chars[rand @chars] for 1..8;
+	return $key;
+}
+
+###
+# setup
+
+# reset audit
+system("auditctl -D >& /dev/null");
+
+# create stdout/stderr sinks
+(my $fh_out, my $stdout) = tempfile(TEMPLATE => '/tmp/audit-testsuite-out-XXXX',
+				    UNLINK => 1);
+(my $fh_err, my $stderr) = tempfile(TEMPLATE => '/tmp/audit-testsuite-err-XXXX',
+				    UNLINK => 1);
+(my $fh_out2, my $stdout2) = tempfile(
+				TEMPLATE => '/tmp/audit-testsuite-out-XXXX',
+				UNLINK => 1);
+(my $fh_err2, my $stderr2) = tempfile(
+				TEMPLATE => '/tmp/audit-testsuite-err-XXXX',
+				UNLINK => 1);
+
+###
+# tests
+
+# set the socketcall filter
+my $key = key_gen();
+system("auditctl -a always,exit -F arch=b64 -S clone -S close -S creat -S dup -S dup2 -S dup3 -S execve -S exit -S exit_group -S fork -S open -S openat -S rename -S renameat -S unlink -S unlinkat -S vfork -S 288 -S accept -S connect -S listen -S socket -S socketpair -k $key");
+system("auditctl -a always,exit -F arch=b32 -S clone -S close -S creat -S dup -S dup2 -S dup3 -S execve -S exit -S exit_group -S fork -S open -S openat -S rename -S renameat -S unlink -S unlinkat -S vfork -k $key");
+# connect
+system("auditctl -a always,exit -F arch=b32 -S socketcall -F a0=3 -k $key");
+# listen
+system("auditctl -a always,exit -F arch=b32 -S socketcall -F a0=4 -k $key");
+# accept
+system("auditctl -a always,exit -F arch=b32 -S socketcall -F a0=5 -k $key");
+# socketpair
+system("auditctl -a always,exit -F arch=b32 -S socketcall -F a0=8 -k $key");
+# accept4
+system("auditctl -a always,exit -F arch=b32 -S socketcall -F a0=18 -k $key");
+
+# run the test
+system("echo Hello World. | $basedir/conn localhost 22");
```

Comment:
In fact, there must be a perl library that can be called from the test script that can listen on a high port.  Since we need perl-Test, perl-Test-Harness, perl-File-Which anyways, add a socket daemon helper...   and connect directly from the script itself using potentially another perl lib.